src/odemis/acq/test/fastem_test.py

```diff
@@ -535,6 +543,52 @@ def test_get_abs_stage_movement(self):
             actual_position = task.get_abs_stage_movement()  # [m]
             numpy.testing.assert_allclose(actual_position, expected_position)
 
+    def test_pre_calibrate(self):
+        """
+        Test the correct stage positions are returned for the corner fields of
+        ROAs consisting of a varying number of single fields.
+        """
+        res_x, res_y = self.multibeam.resolution.value  # single field size
+        px_size_x, px_size_y = self.multibeam.pixelSize.value
+
+        x_fields = 5
+        y_fields = 8
+
+        # The coordinates of the ROA in meters.
+        xmin, ymin, xmax, ymax = (0, 0, res_x * px_size_x * x_fields, res_y * px_size_y * y_fields)
+        coordinates = (xmin, ymin, xmax, ymax)  # in m
+
+        # Create an ROA with the coordinates of the field.
+        roa_name = time.strftime("test_megafield_id-%Y-%m-%d-%H-%M-%S")
+        roa = fastem.FastEMROA(roa_name, coordinates, None,
+                               self.asm, self.multibeam, self.descanner,
+                               self.mppc)
+
+        task = fastem.AcquisitionTask(self.scanner, self.multibeam, self.descanner,
+                                      self.mppc, self.stage, self.ccd,
+                                      self.beamshift, self.lens,
+                                      roa, path=None, future=None, pre_calibrate=True)
+
+        self.descanner.updateMetadata({model.MD_SCAN_GAIN: (5000, 5000)})
+
+        task.pre_calibrate = MagicMock(name='method')
+        task.pre_calibrate.return_value = None
+
+        scan_offset = (0.5, 0.5)
```

Comment:
I wonder if we should use values that also make sense when you run this test on real HW, so that the calibration still works.